### DISCUSSION TITLE
Microsoft Dependency Injection instead of Autofac / Hosted Services instead of Quartz

### DISCUSSION THREAD

**Discussion by ghost** [kgrzybek/modular-monolith-with-ddd#328] [General]
Hello @kgrzybek .. First of all thank you very much for the project. I actually have couple of questions

1) Why Autofac was preferred instead of Microsoft Dependency Injection framework which is part of .Net.
2) Why Quartz.net was preferred instead of Hosted Services that is part of .Net framework?

Thanks.

**Comment by nefosfatus** [comment]
Hello, @isaranghi,

I'll try to answer your first question. Firstly, the choice of tools often depends on the developer's preferences. Secondly, this repository has a long history, and at its inception, Autofac was a more advanced option compared to Microsoft's Dependency Injection.

Lastly, in my opinion, the following construction is very elegant:
```
var infrastructureAssembly = typeof(Context).Assembly;
builder.RegisterAssemblyTypes(infrastructureAssembly)  
    .Where(type => type.Name.EndsWith("Repository"))  
    .AsImplementedInterfaces()  
    .InstancePerLifetimeScope();
```
